torch/_subclasses/meta_utils.py

```diff
@@ -165,36 +137,42 @@ def del_ten():
                 return
             # on shutdown, tensor_ref_key may not be in memo
             self_ref.tensor_memo.pop(tensor_ref_key, None)
-            if weak_st and weak_st.expired():
-                self_ref.storage_memo.pop(weak_st, None)
-            elif weak_st is not None:
-                # [expired-storages]
-                # NB: even though the tensor has died,
-                # the deallocation of its storage can take longer,
-                # even when the storage has no other uses/views.
-                # In this case, the StorageWeakRef object will be kept alive
-                # longer than it needs to be, however the storage itself
-                # will be deallocated. We retain the possibly dead storages
-                # and periodically check if any of them are expired and
-                # can be freed.
-                self_ref.maybe_storages_to_delete.append(weak_st)
 
         weakref.finalize(t, del_ten)
         self.tensor_memo[tensor_ref_key] = v
 
+    def get_storage_memo(self, s):
+        return self.storage_memo.get(WeakIdRef(s), None)
+
+    def set_storage_memo(self, s, v):
+        # hold a weak ref to self, otherwise it will be kept alive
+        # by the del_ten closure
+        self_weak_ref = weakref.ref(self)
+        storage_ref_key = WeakIdRef(s)
+
+        def del_storage():
```

Comment:
should this be factored out with `del_tensor` ?

Reply:
This gets simplified EVEN MORE in the next PR, so just leaving as is.